src/internal/analytics-metadata/__tests__/dom-utils.test.tsx

```diff
@@ -0,0 +1,106 @@
+  test('finds parent', () => {
+    const { container } = render(
+      <div id="parent">
+        <div id="child"></div>
+      </div>
+    );
+    const child = container.querySelector('#child');
+    expect(findNextNode(child as HTMLElement)?.id).toEqual('parent');
+  });
+  test('returns null when child does not exist', () => {
+    const { container } = render(
+      <div id="parent">
+        <div id="child"></div>
+      </div>
+    );
+    const child = container.querySelector('#wrong-child');
+    expect(findNextNode(child as HTMLElement)).toBeNull();
+  });
+  test('returns null when parent does not exist', () => {
+    const { container } = render(
+      <div id="parent">
+        <div id="child"></div>
+      </div>
+    );
+    expect(findNextNode(container.parentElement?.parentElement as HTMLElement)).toBeNull();
+  });
+  test('returns element referred to with referrerId', () => {
+    const { container } = render(
+      <div>
+        <div id="child" data-awsui-referrer-id="id:rr5:"></div>
+        <div id="id:rr5:" className="parent"></div>
+      </div>
+    );
+    const child = container.querySelector('#child');
+    expect(findNextNode(child as HTMLElement)?.classList.contains('parent')).toBe(true);
+  });
+  test('returns null when element referred to with referrerId does not exist', () => {
+    const { container } = render(
+      <div>
+        <div id="child" data-awsui-referrer-id="id:rr5:"></div>
+        <div id="id:rr5:wrong" className="parent"></div>
+      </div>
+    );
+    const child = container.querySelector('#child');
+    expect(findNextNode(child as HTMLElement)).toBeNull();
+  });
+});
+
+describe('isNodeComponent', () => {
+  test('returns false when element has no component metadata', () => {
+    const { container } = render(<div />);
+    expect(isNodeComponent(container.firstElementChild as HTMLElement)).toBe(false);
+  });
+  test('returns true when element has component metadata with component name', () => {
+    const { container } = render(<div {...getAnalyticsMetadataAttribute({ component: { name: 'ComponentName' } })} />);
+    expect(isNodeComponent(container.firstElementChild as HTMLElement)).toBe(true);
+  });
+  test('returns false when element has component metadata without component name', () => {
+    const { container } = render(<div {...getAnalyticsMetadataAttribute({ component: { label: 'labelSelector' } })} />);
+    expect(isNodeComponent(container.firstElementChild as HTMLElement)).toBe(false);
+  });
+  test('returns false when element has corrupted metadata', () => {
+    const { container } = render(<div {...{ [METADATA_ATTRIBUTE]: "{'corruptedJSON':}" }} />);
+    expect(isNodeComponent(container.firstElementChild as HTMLElement)).toBe(false);
+  });
+});
+
+describe('findComponentUp', () => {
+  test('returns null when input is null', () => {
+    expect(findComponentUp(null)).toBeNull();
+  });
+  test('returns parent component element', () => {
```

Comment:
Missing a test for if it has a parent node but no parent component

```
<div id="component-element"> 
   <div id="target-element"></div>
</div>
```

```
expect(findComponentUp(container.querySelector('#target-element')).toBe(null);
```

Reply:
I modified an existing test to cover this case